### PR TITLE
Closes #1981 - Sphinx Utilization Updates

### LINKED ISSUE #1981: Sphinx Utilization Updates
Currently, we used Sphinx to generate our GitHub Pages. One thing I have noticed is that there is a fair bit of duplication of documentation between our GitHub Pages and our markdown files in the repo. We need to look into reducing our documentation maintenance cost by removing the duplication. I have listed out the steps we will need to take below.

1. Look into using markdown to generate Sphinx documentation.
2. Configure documentation in `pydoc` to link out to the markdown files in the root of the repository.
3. Remove any outdated documentation
4. Reorganize our documentation to make locating information easier
5. Link markdown files to the appropriate place in GitHub Pages.

Ultimately, I would like to see our documentation mostly reside in GitHub pages (developer and user). This way we have a consistent and searchable documentation base that would not be duplicated depending on the location or outdated depending on the location. Since users have mentioned some of the outdated documentation recently, we should look into this relatively soon.

### PR DESCRIPTION
**Pull request by Ethan-DeBandi99**
Closes #1981

This PR is a fairly major overhaul of our documentation. It moves a lot of documentation into `pydoc` in order to build our GitHub Pages with Sphinx. The changes are listed out below:

- Updated the theme to `furo`. The theme we updated to previously does not properly render bulleted lists. This theme does and looks much better than our original theme.
- Adds `myst-parser` and `linkify-it-py` dependencies. These are required for properly parsing Markdown files into rst with Sphinx.
- Updates the `pydoc/index.rst` file to remove unnecessary links. 
- Updates the menu based on alterations to documents. 
- Deletes the `CHANGES.md` file. This has been replaced by our release notes.
- Links `CONTRIBUTING.md` into GitHub Pages for more exposure on how to contribute to Arkouda. This required modification to the formatting of the file, but the information remained the same.
- Moves `ENVIRONMENT.md` into `pydoc`
- Deletes `pydoc/contributing.rst`, `pydoc/setup/prerequisites.rst`, and `pydoc/setup/installation.rst`
- Adds an install Landing page and links to Markdown files for specific OS installs.
- Updated links in `README.md` for files that were moved.

Future work that was not handled in this PR:
- Updates to `FILEIO.md`. This file is very out of date and does not contain information on Parquet yet. I moved the file, but will be adding an Issue for updating it separately. (Issue #1986)
- Updates to the documents created using the `autoapi`. This mostly lives in `pydoc/usage`, but should be reviewed to ensure nothing crucial is missing or outdated. (Issue #1987)
- Update `quickstart` to have downloadable Arkouda file and link to chapel with minimal build instructions. (Issue #1988)

**I would recommend pulling down this PR and running `make doc` and then opening `docs/index.html` in your browser to view. You will need to `pip install furo`, `pip install myst-parser`, and `pip install linkify-it-py` first.**

I am posting this as a draft first to make sure everyone is on board with the formatting of the files and the directory.

### CODE REVIEW

File: pydoc/setup/LINUX_INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+# It is recommended to add these variables to a ~/.chplconfig file to prevent having 
+# to export them again
+export CHPL_COMM=gasnet
+# If you're going to be running locally only, use CHPL_COMM=none instead to improve build time
+export CHPL_COMM_SUBSTRATE=smp
+export CHPL_TARGET_CPU=native
+export GASNET_QUIET=Y
+export CHPL_RT_OVERSUBSCRIBED=yes
+export CHPL_RE2=bundled
+export CHPL_LLVM=bundled
```
Comment (pierce314159) [inline pydoc/setup/LINUX_INSTALL.md:33]: we need to make sure to add `CHPL_GMP=bundled` for the bigint functionality

File: pydoc/setup/MAC_INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+source util/quickstart/setchplenv.bash
+```
+
+5) Update some environment variables to the recommended settings. *If you have not installed LLVM, set `CHPL_LLVM=bundled`. It is recommended to install LLVM using Homebrew, `brew install llvm`.* 
+
+6) Add the following to your `rc` file.
+
+```bash
+export CHPL_LLVM=system
+export CHPL_RE2=bundled
```
Comment (pierce314159) [inline pydoc/setup/MAC_INSTALL.md:39]: we should note that `CHPL_GMP` needs to be set to whatever `CHPL_LLVM` is set to

File: pydoc/FILEIO.md [hunk 3]
```diff
@@ -84,16 +85,19 @@ While most objects in Arkouda can be saved, there are 3 main datatypes currently
...
 
 ## Parquet
+
 COMING SOON
 
 ## Reading Objects
+
 Arkouda objects can be read from files using the `ak.read()` or `ak.load()` functions. More information on these functions are linked below.
 
 - [ak.read](https://bears-r-us.github.io/arkouda/autoapi/arkouda/pdarrayIO/index.html#arkouda.pdarrayIO.read)
```
Comment (pierce314159) [inline pydoc/FILEIO.md:95]: `ak.read` has been removed so this link gives a 404. So does `save_all` below

**Review by pierce314159 — COMMENTED** [review]
Couple small things, but wayyyy better overall! Nice work!!

Also I wanted to note the chapel API points to a blank page which is better than 404 like it used to do. But we should prob add that to future work

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -117,23 +106,26 @@ See our wiki for more info on how to run our tests and create your own:
...
+
 Every pull request should have at least one associated issue (if there's not one, create one!).
- 
+
 The issue number(s) should be listed in the title and the body using [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
 Using github markdown (especially [code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
 is very appreciated. When in doubt, take a look at some closed pull requests.
 
-````
+```text
```
[the remark this answers is not in the document]
Comment (Ethan-DeBandi99) [reply CONTRIBUTING.md:128]: Yeah I see that now. I was just going through and fixing everything that was noted as a syntax error. Updated this to match previous.

File: pydoc/setup/MAC_INSTALL.md (hunk 2 above)
Comment (Ethan-DeBandi99) [reply pydoc/setup/MAC_INSTALL.md:39]: Updated

File: pydoc/setup/LINUX_INSTALL.md (hunk 1 above)
Comment (Ethan-DeBandi99) [reply pydoc/setup/LINUX_INSTALL.md:33]: Added

**Review by pierce314159 — APPROVED** [review]
Looks good!

**Review by joshmarshall1 — APPROVED** [review]
This looks a lot better than what we had before. Only one tiny thing I found that may just be on my end.

**Review by jaketrookman — APPROVED** [review]
Looks good

**Pull request merged**